core/src/labels/label.cpp

```diff
@@ -228,32 +219,37 @@ void Label::pushTransform() {
 
 void Label::resetState() {
     m_occludedLastFrame = false;
-    m_skipTransitions = false;
-    m_occlusionSolved = false;
+    m_occluded = false;
     m_updateMeshVisibility = true;
     m_dirty = true;
     m_proxy = false;
     enterState(State::wait_occ, 0.0);
 }
 
-bool Label::updateState(const glm::mat4& _mvp, const glm::vec2& _screenSize, float _dt, float _zoomFract) {
-    if (m_currentState == State::dead) {
+bool Label::update(const glm::mat4& _mvp, const glm::vec2& _screenSize, float _zoomFract) {
+    if (m_state == State::dead) {
         return false;
     }
 
-    bool occludedLastFrame = m_occludedLastFrame;
-    m_occludedLastFrame = false;
+    m_occluded = false;
 
-    bool ruleSatisfied = updateScreenTransform(_mvp, _screenSize);
+    bool ruleSatisfied = updateScreenTransform(_mvp, _screenSize, true);
 
     // one of the label rules has not been satisfied
     if (!ruleSatisfied) {
-
-        // go to dead state, this breaks determinism, but reduce potential label set since a lot
-        // of discarded labels are discared for line exceed (lots of tiny small lines on a curve
-        // for example), which won't have their rule satisfied
-        enterState(State::dead, 0.0);
+        if ((m_state & State::wait_occ) != 0) {
```

Comment:
Say a tile is initially shown at tile scale ~2. Line labels that just fit on the line 'satisfy the rule' and wait_occ may change to visible state. Then one slightly zooms out so that the rule is not satisfied - now when zooming back in again the label should also be able to show up again.